source/github-helpers/index.ts

```diff
@@ -98,6 +98,9 @@ const escapeRegex = (string: string): string => string.replace(/[\\^$.*+?()[\]{}
 const prCommitPathnameRegex = /[/][^/]+[/][^/]+[/]pull[/](\d+)[/]commits[/]([\da-f]{7})[\da-f]{33}(?:#[\w-]+)?\b/; // eslint-disable-line unicorn/better-regex
 export const prCommitUrlRegex = new RegExp('\\b' + escapeRegex(location.origin) + prCommitPathnameRegex.source, 'gi');
 
+const prComparePathnameRegex = /[/][^/]+[/][^/]+[/]compare[/]([v\da-f.]{7})...(master|[v\da-f.]{7})#?diff-([\da-f]{7})[\da-f]{25}(?:[#\d\w-]+)?\b/; // eslint-disable-line unicorn/better-regex
```

Comment:
The regex is incorrect. You can compare any branches and commits, not just `v*` tags and `master`

Probably a generic `(.+)` is enough to capture the entire comparison
```diff
- ([v\da-f.]{7})...(master|[v\da-f.]{7})
+ (.+)
```

The `#` **must** be there, so

```diff
- #?diff
+ #diff
```

I think the `(?:[#\d\w-]+)?` part refers to the URL hash, which in this case is already `#diff-*` so it should be dropped

Reply:
https://github.com/sindresorhus/refined-github/compare/20.9.11...masterdiff-67ba01fa3e792891b585fa43e92c78ea

This link doesn't have a hash

Reply:
Re-read the issue that you’re trying to fix. That link does not have the hash… because it’s broken. This feature is meant to avoid such broken link.

Reply:
Even the “test URLs” you provided don’t make sense here. This feature acts in the comment box, not on those URLs. Refer to https://github.com/sindresorhus/refined-github/pull/3223 to see how this feature should be tested.